### PR TITLE
Fixes portals teleporting anchored things [!!INCLUDING MEGAFAUNA!!] and observers not being able to go through them by clicking on them.

### PR DESCRIPTION
**Pull request by silicons**

### CODE REVIEW

**Comment by SpaceManiac** [comment]
Not sure why the existing isn't working, but it should probably be updated rather than awkwardly overridden. https://github.com/tgstation/tgstation/blob/7a3a66a4f8b8bc5c8e993d5e1b9bd61df90d36a5/code/_onclick/observer.dm#L69

**Comment by kevinz000** [comment]
kek I wasn't even aware there was an existing.

File: code/game/objects/effects/portals.dm [hunk 1]
```diff
@@ -124,12 +127,18 @@
...
+		return ..()
+
 /obj/effect/portal/proc/teleport(atom/movable/M)
-	if(!istype(M) || istype(M, /obj/effect) || isobserver(M) || (ismecha(M) && !mech_sized) || (!isobj(M) && !ismob(M))) //Things that shouldn't teleport.
+	if(!istype(M) || istype(M, /obj/effect) || (ismecha(M) && !mech_sized) || (!isobj(M) && !ismob(M))) //Things that shouldn't teleport.
 		return
 	var/turf/real_target = get_link_target_turf()
 	if(!istype(real_target))
 		return FALSE
+	if(M.anchored && !allow_anchored)
```
Comment (ChangelingRain) [inline code/game/objects/effects/portals.dm]: mechs are anchooooored. i swear people keep trying to do this, but mechs are anchored.

Comment (kevinz000) [reply code/game/objects/effects/portals.dm]: fixing soon!

**Pull request closed without merging**